browser/tool.gpt

```diff
@@ -0,0 +1,122 @@
+Params: tabID: (optional) The ID of the tab. If unspecified, a new tab will be created, and its ID will be returned.
+
+#!http://service.daemon.gptscript.local/browse
+
+---
+Name: Get Page Contents
+Share Context: Browser Context
+Tools: service
+Description: Returns the content of a website in Markdown format.
+Params: website: (optional) The HTTPS URL of the website to visit. If unspecified, the current tab will be used.
+Params: tabID: (optional) The ID of the tab. If unspecified, a new tab will be created.
+
+#!http://service.daemon.gptscript.local/getPageContents
+
+---
+Name: Filter
+Share Context: Browser Context
+Tools: service
+Description: filter the page to get elements based on specific id, html tag, or class.
+Params: filter: (required) the class (eg. '.foo') or id (eg. '#foo') of an object.
+Params: tabID: (required) The ID of the tab to filter contents of.
+Tools: service
+
+#!http://service.daemon.gptscript.local/getFilteredContent
+
+---
+Name: Fill
+Share Context: Browser Context
+Tools: service
+Description: Fills text into an element on the web page. Useful for filling out forms and other input fields.
+Params: userInput: (required) Description of the element to fill. This will be used by the AI to determine which element to pick from the results. Be as detailed as possible.
+Params: keywords: (required) Comma-separated list of keywords related to the element. These will be used to search for the element on the page. ALWAYS include at least one keyword.
+Params: matchTextOnly: (optional, default false) Matches elements based on their text content and ignores their attributes. Useful for cases where the user has provided an exact piece of text that they want to interact with on the page.
+Params: content: The text to fill into the element.
+Params: tabID: (required) The ID of the tab.
+Tools: service
+
+#!http://service.daemon.gptscript.local/fill
+
+---
+Name: Enter
+Share Context: Browser Context
+Tools: service
+Description: Presses the enter key. Useful after filling out a form or other input.
+Params: tabID: (required) The ID of the tab.
+Tools: service
+
+#!http://service.daemon.gptscript.local/enter
+
+---
+Name: Scroll
+Share Context: Browser Context
+Tools: service
+Description: Scrolls to the bottom of the page, possibly loading more content. Useful for sites like Reddit which support infinite scroll.
+Params: tabID: (required) The ID of the tab.
+Tools: service
+
+#!http://service.daemon.gptscript.local/scrollToBottom
+
+---
+Name: Back
+Share Context: Browser Context
+Tools: service
+Description: Navigates to back to the previous site in history.
+Params: tabID: (required) The ID of the tab.
+Tools: service
+
+#!http://service.daemon.gptscript.local/back
+
+---
+Name: Forward
+Share Context: Browser Context
+Tools: service
+Description: Navigates to forward to the next site in history.
+Params: tabID: (required) The ID of the tab.
+Tools: service
+
+#!http://service.daemon.gptscript.local/forward
```

Comment:
A lot of the tools here aren't useful on their own unless they share pretty much all of the other tools.

What would folks think if I tag everything but the bundle and `Get Page Content` with `Metadata: index: false`?

Reply:
I think that makes sense.

Reply:
done